tests/integration/animation-complete/animationComplete.js

```diff
@@ -266,4 +266,164 @@
+						"Animation context is correct" );
+					start();
+				}
+			};
+
+		returnValue = transitionTest
+			.addClass( "ui-panel-animate ui-panel-position-left ui-panel-display-overlay" )
+			.animationComplete( function() {
+				transitionContext = this;
+				maybeAssert();
+			}, "transition" );
+
+		deepEqual( returnValue, transitionTest,
+			"Returned jQuery object for transition is the one passed in" );
+
+		returnValue = animationTest
+			.addClass( "in" )
+			.animationComplete( function() {
+				animationContext = this;
+				maybeAssert();
+			});
+
+		deepEqual( returnValue, animationTest,
+			"Returned jQuery object for animation is the one passed in" );
+
+	});
+
+	module( "Callback context and return value: fallback" );
+
+	asyncTest( "Make sure context and return value is correct for fallback", function() {
+		expect( 4 );
+		var transitionContext, animationContext, returnValue,
+			transitionTest = $( "#transition-test" ),
+			animationTest = $( "#animation-test" ),
+			completeCount = 0,
+			maybeAssert = function() {
+				completeCount++;
+				if ( completeCount === 2 ) {
+					deepEqual( transitionContext, $( "#transition-test" )[ 0 ],
+						"Transition context is correct" );
+					deepEqual( animationContext, $( "#animation-test" )[ 0 ],
+						"Animation context is correct" );
+					start();
+				}
+			};
+
+		returnValue = transitionTest.animationComplete( function() {
+			transitionContext = this;
+			maybeAssert();
+		}, "transition" );
+
+		deepEqual( returnValue, transitionTest,
+			"Returned jQuery object for transition is the one passed in" );
+
+		returnValue = animationTest.animationComplete( function() {
+			animationContext = this;
+			maybeAssert();
+		});
+
+		deepEqual( returnValue, animationTest,
+			"Returned jQuery object for animation is the one passed in" );
+	});
+
+	module( "Callback context and return value: no support", {
+		setup: function() {
+			oldTransitions = $.support.cssTransitions,
+			oldAnimations = $.support.cssAnimations;
+
+			$.support.cssAnimations = false;
+			$.support.cssTransitions = false;
+		},
+		teardown: function() {
+			$.support.cssTransitions = oldTransitions;
+			$.support.cssAnimations = oldAnimations;
+			$( "#transition-test" )
+				.removeClass( "ui-panel-animate ui-panel-position-left ui-panel-display-overlay" );
+			$( "#animation-test" ).removeClass( "in" );
+		}
+	});
+
+	asyncTest( "Make sure context and return value is correct for no support", function() {
+		expect( 4 );
+		var transitionContext, animationContext, returnValue,
+			transitionTest = $( "#transition-test" ),
+			animationTest = $( "#animation-test" ),
+			completeCount = 0,
+			maybeAssert = function() {
+				completeCount++;
+				if ( completeCount === 2 ) {
+					deepEqual( transitionContext, $( "#transition-test" )[ 0 ],
+						"Transition context is correct" );
+					deepEqual( animationContext, $( "#animation-test" )[ 0 ],
+						"Animation context is correct" );
+					start();
+				}
+			};
```

Comment:
`maybeAssert` is repeated 3 times lets pull this out into the parent scope and make it take arguments.